System/Posix/Env/ByteString.hsc

```diff
@@ -116,15 +133,25 @@ foreign import capi unsafe "HsUnix.h unsetenv"
    c_unsetenv :: CString -> IO ()
 # endif
 #else
-unsetEnv name = putEnv (name ++ "=")
+unsetEnv name = putEnv (name <> BC.pack "=")
 #endif
 
 -- |'putEnv' function takes an argument of the form @name=value@
 -- and is equivalent to @setEnv(key,value,True{-overwrite-})@.
 
 putEnv :: ByteString {- ^ "key=value" -} -> IO ()
-putEnv keyvalue = B.useAsCString keyvalue $ \s ->
-  throwErrnoIfMinus1_ "putenv" (c_putenv s)
+putEnv (PS fp _ l) = withForeignPtr fp $ \p -> do
```

Comment:
Is this code really going to be built against ancient `bytestring` versions that have `PS` instead of `BS`?  And if it were, it would be *very wrong* to ignore the offset.  Use `BS` instead.

Reply:
Using `BS` would require `build-depends: bytestring >= 0.11`. That's acceptable, but it's easy to be more conservative with 
```haskell
putEnv (PS fp o l) = withForeignPtr fp $ \p -> do ... (p `plusPtr` o)
```

Reply:
Sorry... it was almost 2 am at night.